### ISSUE TITLE
cabal init doesn't autodetect author name and maintainer email

### ISSUE THREAD

**Issue by fgaz** [haskell/cabal#8255]
**Describe the bug**

cabal init doesn't autodetect author name and maintainer email. the old init implementation did, and i found it pretty useful

**To Reproduce**

```
$ cabal init
[...]
Author name? [optional] 
Maintainer email? [optional] 
[...]
```

**Expected behavior**

```
[...]
Author name? [default: Name Surname] 
Maintainer email? [default: user@example.com]
[...]
```

(detected from git iirc)

**System information**
 - cabal HEAD
 - NixOS, but my git config is in the usual place

**Additional context**

cc @ptkato @emilypi

**Comment by ptkato** [comment]
It should? Previously it used git, darcs and some weird env flags to try to guess the name/email, now it only uses git.

https://github.com/haskell/cabal/blob/0a2e68cbde9ad8dafd67c18511e7117956060f9b/cabal-install/src/Distribution/Client/Init/NonInteractive/Heuristics.hs#L171-L186

**Comment by fgaz** [comment]
Calling those functions in the repl correctly shows my info. Weird! Maybe that information doesn't actually get used as default?

```
λ import Distribution.Client.Init.NonInteractive.Heuristics 
λ guessAuthorName 
"Francesco Gazzetta"
```

**Comment by fgaz** [comment]
`cabal init -n` works. my guess is that the empty input in the interactive init gets interpreted as `Flag ""` instead of `NoFlag` and overrides the  `Flag "Name from git"` when mappended

**Comment by fgaz** [comment]
Also it'd be nice to display `[default: ...]` like old init

**Comment by ulysses4ever** [comment]
@Mikolaj at least this one should be fixed in #8267
